src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractSerializer.cs

```diff
@@ -34,6 +34,22 @@ public sealed class DataContractSerializer : XmlObjectSerializer
         private ISerializationSurrogateProvider _serializationSurrogateProvider;
         private bool _serializeReadOnlyTypes;
 
+        private static SerializationOption _option = SerializationOption.CodeGenOnly;
+        private static bool _optionAlreadySet;
+        public static SerializationOption Option
+        {
+            get { return _option; }
+            set
+            {
+                if (_optionAlreadySet)
+                {
+                    throw new InvalidOperationException("Can only set once");
+                }
+                _optionAlreadySet = true;
+                _option = value;
+            }
+        }
```

Comment:
Can we add two helper properties as below and change those conditions using SerializationOption to use the properties instead? IMO this would improve the readability of those conditions.

```
public static bool CodeGenEnabled
{
    get
    {
        return Option != SerializationOption.ReflectionOnly;
    }
}

public static bool ReflectionEnabled
{
    get
    {
        return Option != SerializationOption.CodeGenOnly;
    }
}
```